code/iaas/logic/src/main/java/io/cattle/platform/process/storagepool/StoragePoolPurge.java

```diff
@@ -0,0 +1,36 @@
+package io.cattle.platform.process.storagepool;
+
+import io.cattle.platform.core.constants.CredentialConstants;
+import io.cattle.platform.core.constants.StoragePoolConstants;
+import io.cattle.platform.core.model.Credential;
+import io.cattle.platform.core.model.StoragePool;
+import io.cattle.platform.engine.handler.HandlerResult;
+import io.cattle.platform.engine.process.ProcessInstance;
+import io.cattle.platform.engine.process.ProcessState;
+import io.cattle.platform.process.base.AbstractDefaultProcessHandler;
+
+import javax.inject.Named;
+
+@Named
+public class StoragePoolPurge extends AbstractDefaultProcessHandler {
+
+
+    @Override
+    public HandlerResult handle(ProcessState state, ProcessInstance process) {
+        StoragePool registry = (StoragePool) state.getResource();
+        if (registry.getKind().equalsIgnoreCase(StoragePoolConstants.KIND_REGISTRY)){
+
+            for (Credential cred : getObjectManager().children(registry, Credential.class)) {
```

Comment:
@hibooboo2 It makes sense to use `objectManager.find()` so that you can build the removed not null check directly into the sql query.